components/entity-access-rules/api/src/main/java/org/phenotips/data/permissions/internal/DefaultEntityPermissionsPreferencesManager.java

```diff
@@ -0,0 +1,349 @@
+        // Guest user or invalid entity
+        if (currentUserRef == null) {
+            return Collections.emptyMap();
+        }
+
+        Map<EntityReference, Collaborator> defaultCollabsMap = getDefaultPreferenceCollaborators(currentUserRef);
+        // here null value will indicate the absence of any defaultCollaborator properties stored
+        if (defaultCollabsMap != null) {
+            return defaultCollabsMap;
+        }
+
+        DocumentReference workGroupRef = getWorkgroupForUser(currentUserRef);
+        // if the entity is not a user or user has no groups, we can not get further
+        if (workGroupRef != null) {
+            return getDefaultPreferenceCollaborators(workGroupRef);
+        }
+
+        return Collections.emptyMap();
+    }
+
+    @Override
+    public Visibility getDefaultVisibility(DocumentReference entity)
+    {
+        try {
+            DocumentReference currentUserRef = entity == null ? this.helper.getCurrentUser() : entity;
+            // Guest user or invalid entity
+            if (currentUserRef == null) {
+                return null;
+            }
+
+            String visibilityName = getDefaultPreferenceStringValue(currentUserRef, DEFAULT_VISIBILITY_PROPERTY_NAME);
+            if (visibilityName != null) {
+                Visibility defaultVisibility =
+                    this.componentManager.get().getInstance(Visibility.class, visibilityName);
+                if (defaultVisibility != null) {
+                    return defaultVisibility;
+                }
+            }
+
+            DocumentReference workGroupRef = getWorkgroupForUser(currentUserRef);
+            // if the entity is not a user or user has no groups, we can not get further
+            if (workGroupRef == null) {
+                return null;
+            }
+
+            visibilityName = getDefaultPreferenceStringValue(workGroupRef, DEFAULT_VISIBILITY_PROPERTY_NAME);
+            if (visibilityName != null) {
+                Visibility defaultVisibility =
+                    this.componentManager.get().getInstance(Visibility.class, visibilityName);
+                return defaultVisibility;
+            }
+        } catch (ComponentLookupException ex) {
+            this.logger.error("Failed to get the default visibility for entity [{}]: {}", entity.toString(),
+                ex.getMessage(), ex);
+        }
+
+        return null;
+    }
+
+    @Override
+    public DocumentReference getDefaultStudy(DocumentReference entity)
+    {
+        DocumentReference currentUserRef = entity == null ? this.helper.getCurrentUser() : entity;
+        // Guest user or invalid entity
+        if (currentUserRef == null) {
+            return null;
+        }
+
+        String studyID = getDefaultPreferenceStringValue(currentUserRef, DEFAULT_STUDY_PROPERTY_NAME);
+        if (studyID != null) {
+            return this.userOrGroupResolver.resolve(studyID);
```

Comment:
Same comment about using something called `userOrGroupResolver` to resolve an entity that is not supposed to be a user or a group.